### ISSUE TITLE
Add possibility to specify dotfiles configuration at the repository/project level

### ISSUE THREAD

**Issue by NikiforovAll** [microsoft/vscode-remote-release#3715]
I would like to be able to specify dotfile configuration for a particular repository/dev-container, so it is possible to have multiple dotfile definitions per user. 

Also, it is possible to share dotfile "profile" with other people. 

Generally, you don't always want this information specified in `devcontainer.json` because if you have dev-container checked into git, this setting will end up in source control.  I suggest to add an additional file `devcontainer.dotfiles.json`

**Comment by PavelSosin-320** [comment]
@NikiforovAll Multiple configurations is great but sharing profile? If you involved in the 3 / 4 line customer support your customer will be very unhappy if you will share provided for the issue-solving credentials stored in the DotFolder with your colleagues in the same room.  Even when I use git I would like to know who broke the master branch personally. So, I never share my Profile containing dot folders with anybody.

**Comment by NikiforovAll** [comment]
@PavelSosin-320 Yes, indeed. But I think you don't want to add sensitive information anyway. Also, git configuration is overridden by host machine git configuration.

I agree, but still think it would be nice to specify location of dotfiles on project level. E.g. being able to specify config for vsvode workspace.

**Comment by PavelSosin-320** [comment]
I propose to re-phrase this PR and include the requirement to support:
Either Gist that doesn't contain any personal information, only plain text like
[WSL distro installation Gist](https://gist.github.com/artman41/858450e2c29b239d8692213b684ada25)
or Configuration templates with external "values" file like in all modern deployment managers like Ansible, Helm, etc.
But this is a huge topic because json is insufficient for such purposes and interpreter. Maybe something out-of-the-shelf can be used.

**Comment by PavelSosin-320** [comment]
Unfortunately, your assumption that all utilities have a universal --configuration option is weak. The silent agreement between all unixoids is that all dot folders are placed in Users's home folder which created during new user creation by mkuser with the correct access attributes and can't be changed until the user exists. All installers create corresponding dot folders and put default configuration files into this folder. It's impossible to change it. There is another concept of how to create more than one profile - workspace. It is possible to switch users when the workspace is switched. It is called multitenancy and used only in big distributed multiuser systems due to high development and resource cost.
